evennia/server/portal/webclient.py

```diff
@@ -73,8 +73,13 @@ def onOpen(self):
         This is called when the WebSocket connection is fully established.
 
         """
-        client_address = self.transport.client
-        client_address = client_address[0] if client_address else None
+        if 'x-forwarded-for' in self.http_headers and self.trustXForwardedFor:
+            addresses = [x.strip() for x in self.http_headers['x-forwarded-for'].split(',')]
+            trusted_addresses = addresses[-self.trustXForwardedFor:]
```

Comment:
I think this might be off-by-one.  If the header value passed is 'x-forwarded-for: 192.168.1.1, 127.0.0.1' and you have WEBSOCKET_TRUST_X_FORWARDED_FOR = 1 (meaning 127.0.0.1 is the trusted proxy), then you end up with 127.0.0.1 as the client_address...?

Reply:
It looks like this won't occur, because 127.0.0.1 will never be in the `X-Forwarded-For` header.

According to Wikipedia, the format of `X-Forwarded-For` is:

```
X-Forwarded-For: client, proxy1, proxy2
```

The leftmost IP address is always the original client address, followed by each proxy server that the request passed through, _minus the final proxy server._ I think this solves the off-by-one.

I ran a few test cases with this in mind:

#### Situation 1: a single reverse proxy
```
# Client IP: 192.168.1.1
# Local Nginx Proxy: 127.0.0.1

addresses = ["192.168.1.1"]  # X-Forwarded-For: 192.168.1.1
trust_x_forwarded_for = 1  # We expect 1 proxy server
trusted_addresses = addresses[-trust_x_forwarded_for:]  # ["192.168.1.1"]
client_address = trusted_addresses[0]  # "192.168.1.1"
```

#### Situation 2: reverse proxy and a load balancer
```
# Client IP: 192.168.1.1
# Load Balancer: 140.82.118.4
# Local Nginx Proxy: 127.0.0.1

addresses = ["192.168.1.1", "140.82.118.4"]  # X-Forwarded-For: 192.168.1.1, 140.82.118.4
trust_x_forwarded_for = 2  # We expect 2 proxy servers
trusted_addresses = addresses[-trust_x_forwarded_for:]  # ["192.168.1.1", "140.82.118.4"]
client_address = trusted_addresses[0]  # "192.168.1.1"
```

#### Situation 3: reverse proxy and a 3rd party proxy server
```
# Client IP: 192.168.1.1
# 3rd party proxy server: 140.82.118.4
# Local Nginx Proxy: 127.0.0.1

addresses = ["192.168.1.1", "140.82.118.4"]  # X-Forwarded-For: 192.168.1.1, 140.82.118.4
trust_x_forwarded_for = 1  # We're expecting only our local nginx proxy
trusted_addresses = addresses[-trust_x_forwarded_for:]  # ["140.82.118.4"]
client_address = trusted_addresses[0]  # "140.82.118.4"
```

Reply:
I guess I didn't remember the tidbit about the last proxy being excluded.  Thanks.

Reply:
A code comment here clarifying just how the x-forward-for indexing works here would be useful to improve maintainability of this code (similarly as was explained to friarzen)